src/lib/uTop_main.cppo.ml

```diff
@@ -183,6 +183,11 @@ class read_phrase ~term = object(self)
           let result = parse_and_check input eos_is_error in
           return_value <- Some result;
           LTerm_history.add UTop.history input;
+          ignore(match result with
+            | UTop.Value _, _ ->
+                LTerm_history.add UTop.history input;
```

Comment:
@diml The stashing feature maintains a separate history (it only makes sense to stash lines which compiled without incident). As I understand the code here, the `result` against which I am matching on the highlighted lines will only match with the highlighted pattern if its expression compiled successfully.